src/modules/asab-config/ConfigEditor.js

```diff
@@ -244,8 +250,7 @@ function ConfigSection(props) {
 				<CardBody>
 					{Object.keys(props.section.properties).map((item_name, idx) =>
 						// Decide what type of config item to render based on format
-						// TODO: `format` names should be consulted with BE team
-						{switch(props.section.properties[item_name]['format']){
+						{switch(props.section.properties[item_name]['$defs'] ? Object.keys(props.section.properties[item_name]['$defs'])[0] : undefined){
```

Comment:
The main remark here is that $def should not not be required ... use format since that is always present and $def contains only "extra" configuration required by UI.

Reply:
@ateska `$defs` is required only for generating input formats for UI (it is used for softcheck, if the values are passed to zookeeper in correct form), if it is not defined, the `text` input is used as default as you can see here:

https://github.com/TeskaLabs/asab-webui/blob/refactor/obtaining-input-formats-from-schemas/src/modules/asab-config/ConfigEditor.js#L317-#L323

So if no `$defs` is passed in the JSON schema, the text input field is considered as a default.

As we discussed and if I remember correctly, the input field `format` for UI should be defined in `$defs` because implementing format out of `$defs` would break rules of JSON schema. Maybe instead of `format` you meant `type` ? Because this is present in every  JSON schema.

Here is the example of JSON schema from Mithun. For UI there is no other way to obtain the information what input field to render than `$defs` (or `type`)

```
                "precise_error_handling": {
                    "$id": "#/properties/connection%3AESConnection/properties/precise_error_handling",
                    "type": "boolean",
                    "title": "The precise_error_handling schema",
                    "description": "If set to True, checkbox is checked. Otherwise false.",
                    "default": false,
                    "examples": [
                        true
                    ],
                    "$defs": {
                        "checkbox": {
                          "type": "boolean",
                          "format": "checkbox"
                        }
                    }
                }
```

Reply:
If you meant `type`, I will refactor accordingly and will use `$defs` as an additional info for UI

Reply:
Well, I mean that if the "type" is boolean, I don't need to specify `$defs` in order to use checkbox.

The goal - as usual - is to keep these files as short as possible in majority of cases.

Reply:
Got it! Thank you